graspy/inference/latent_distribution_test.py

```diff
@@ -14,121 +14,158 @@
-                msg = "n_components must an int, not {}.".format(type(n_components))
+                msg = "n_components must be an int, not {}.".format(type(n_components))
                 raise TypeError(msg)
 
         if not isinstance(n_bootstraps, int):
-            msg = "n_bootstraps must an int, not {}".format(type(n_bootstraps))
+            msg = "n_bootstraps must be an int, not {}".format(type(n_bootstraps))
             raise TypeError(msg)
         elif n_bootstraps < 1:
             msg = "{} is invalid number of bootstraps, must be greater than 1"
             raise ValueError(msg.format(n_bootstraps))
 
-        if bandwidth is not None and not isinstance(bandwidth, float):
-            msg = "bandwidth must an int, not {}".format(type(bandwidth))
+        if not isinstance(workers, int):
+            msg = "workers must be an int, not {}".format(type(workers))
             raise TypeError(msg)
+        elif workers <= 0:
+            msg = "{} is invalid number of workers, must be greater than 0"
+            raise ValueError(msg.format(workers))
 
         super().__init__(embedding="ase", n_components=n_components)
+
+        if callable(metric):
+            metric_func = metric
+        else:
+            if metric == "gaussian":
+                metric_func = _medial_gaussian_kernel
+            else:
+
+                def metric_func(X, Y=None, metric=metric, workers=None):
+                    return pairwise_distances(X, Y, metric=metric)
+
+        self.test = KSample(test, compute_distance=metric_func)
         self.n_bootstraps = n_bootstraps
-        self.bandwidth = bandwidth
-
-    def _gaussian_covariance(self, X, Y):
-        diffs = np.expand_dims(X, 1) - np.expand_dims(Y, 0)
-        if self.bandwidth is None:
-            self.bandwidth = 0.5
-        return np.exp(-0.5 * np.sum(diffs ** 2, axis=2) / self.bandwidth ** 2)
-
-    def _statistic(self, X, Y):
-        N, _ = X.shape
-        M, _ = Y.shape
-        x_stat = np.sum(self._gaussian_covariance(X, X) - np.eye(N)) / (N * (N - 1))
-        y_stat = np.sum(self._gaussian_covariance(Y, Y) - np.eye(M)) / (M * (M - 1))
-        xy_stat = np.sum(self._gaussian_covariance(X, Y)) / (N * M)
-        return x_stat - 2 * xy_stat + y_stat
+        self.workers = workers
 
     def _embed(self, A1, A2):
+        if not is_symmetric(A1) or not is_symmetric(A2):
+            raise NotImplementedError()  # TODO asymmetric case
+
+        if self.n_components is None:
+            num_dims1 = select_dimension(A1)[0][-1]
+            num_dims2 = select_dimension(A2)[0][-1]
+            self.n_components = max(num_dims1, num_dims2)
+
         ase = AdjacencySpectralEmbed(n_components=self.n_components)
         X1_hat = ase.fit_transform(A1)
         X2_hat = ase.fit_transform(A2)
+
         if isinstance(X1_hat, tuple) and isinstance(X2_hat, tuple):
             X1_hat = np.concatenate(X1_hat, axis=-1)
             X2_hat = np.concatenate(X2_hat, axis=-1)
```

Comment:
@bdpedigo do you know why is this here if directed inputs aren't supported anyway? or the contrary of why do we not support directed inputs if this is here?

Reply:
was probably a copy pasta from semipar code. I don't know if we want to support directed or not. do you have opinions on this?

Reply:
I'd say we should eventually after we run simulations with directed inputs and verify that the tests have intended behavior. issue for now?

Reply:
issue for now sounds good

Reply:
#358 . resolving this